validation_state.go

```diff
@@ -27,6 +27,9 @@ type ValidationState struct {
 	Misc                        map[string]interface{}
 
 	Errs *[]KeyError
+
+	// ExtraData is a shared storage between all substates of a ValidationState
+	ExtraData *map[string]interface{}
```

Comment:
Let's rename this to `AdditionalValidationData` and its respective comments/function names.

Reply:
> The struct isn't cleared on a new substate and keys can easily collide depending on usage.

Not clearing the struct was the whole point of adding the new field. That is pretty much the difference to the Misc field.

I'll a basic test that at least confirms my usecase and make the name changes